### PR TITLE
fix the functions to wait to the result

### PR DESCRIPTION
**Pull request by sharlotta93**
### **PR Type**
enhancement, bug fix

___

### **Description**
- Enhanced logging in the provider API to include body parsing and error details.
- Fixed asynchronous handling in provider fetching functions to ensure they wait for results.
- Improved logging in the API service to include the number of results returned.
- Updated the `@awell-health/sol-scheduling` dependency to a newer version.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Enhance logging for provider API requests and errors</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pages/api/sol/providers/index.ts

<li>Added logging for body parsing and error handling.<br> <li> Improved error logging when fetching providers fails.<br>

</details>

  </td>
  <td><a href="https://github.com/awell-health/hosted-pages/pull/287/files#diff-a25e848a9831907f0e846538f9deb5f90b25cbc3154e01a4d6ea5e10b00dd6ab">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>api.service.ts</strong><dd><code>Improve logging details in API service</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/components/Extension/PrivateExtensions/actions/IntakeScheduling/api.service.ts

- Updated logging to include the number of results.

</details>

  </td>
  <td><a href="https://github.com/awell-health/hosted-pages/pull/287/files#diff-6155fdaeac382069a749819f730aad0eccc903e4f5d28738c69b578bd7957758">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>IntakeScheduling.tsx</strong><dd><code>Fix asynchronous handling in provider fetching functions</code>&nbsp; </dd></summary>
<hr>

src/components/Extension/PrivateExtensions/actions/IntakeScheduling/IntakeScheduling.tsx

<li>Modified fetch functions to be asynchronous.<br> <li> Ensured functions wait for provider data retrieval.<br>

</details>

  </td>
  <td><a href="https://github.com/awell-health/hosted-pages/pull/287/files#diff-a89a9fac197c0d8893e45e4784f7fab58c59a8cd94e8ac34d5758a8d5ff3c99c">+14/-4</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Update sol-scheduling dependency version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

package.json

- Updated `@awell-health/sol-scheduling` dependency version.

</details>

  </td>
  <td><a href="https://github.com/awell-health/hosted-pages/pull/287/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: pages/api/sol/providers/index.ts [hunk 1]
```diff
@@ -22,6 +22,14 @@ export default async function handler(
     const accessToken = await getAccessToken(omit(settings, 'baseUrl'))
 
     const bodyValidation = GetProvidersInputSchema.safeParse(req.body)
+    log(
+      {
+        message: 'SOL: Parsing body',
+        bodyValidation,
+        body: req.body,
+      },
+      'INFO'
```
Comment (bejoinka) [inline pages/api/sol/providers/index.ts:31]: you shouldn't need info anymore -- that's default behavior

File: src/components/Extension/PrivateExtensions/actions/IntakeScheduling/IntakeScheduling.tsx [hunk 2]
```diff
@@ -57,14 +57,24 @@ export const IntakeScheduling: FC<IntakeSchedulingProps> = ({
   }, [updateLayoutMode, resetLayoutMode])
 
   const fetchProviderFn = useCallback(
-    (providerId: string) =>
-      fetchProvider({ input: { providerId }, requestOptions: { baseUrl } }),
+    async (providerId: string) => {
+      const provider = fetchProvider({
```
Comment (bejoinka) [inline src/components/Extension/PrivateExtensions/actions/IntakeScheduling/IntakeScheduling.tsx:61]: are you supposed to be awaiting this `fetchProvider` function as well?

**Review by bejoinka — CHANGES_REQUESTED** [review]
i assume in one of these you want to return the resolved value instead of the promise, given the change.

File: src/components/Extension/PrivateExtensions/actions/IntakeScheduling/IntakeScheduling.tsx (hunk 2 above)
Comment (sharlotta93) [reply src/components/Extension/PrivateExtensions/actions/IntakeScheduling/IntakeScheduling.tsx:61]: good point!

**Review by bejoinka — APPROVED** [review]

**Pull request merged**